Library/Homebrew/test/test_versions.rb

```diff
@@ -54,8 +54,10 @@ def test_comparing_regular_versions
   end
 
   def test_HEAD
-    assert_operator version("HEAD"), :>, version("1.2.3")
-    assert_operator version("1.2.3"), :<, version("HEAD")
+    assert_operator head_version("HEAD"), :>, version("1.2.3")
+    assert_operator head_version("HEAD-abcdef"), :>, version("1.2.3")
```

Comment:
@MikeMcQuaid  :+1: Helped me to notice that I've forgotten to rewrite `Version` comparator a little, but have been relying on the behaviour when the comparator is rewritten. It doesn't change anything too much if it all worked without that comparator, but anyway.

I suppose that `HEAD` version equals to any other `HEAD` version. If we want to get latest `HEAD` prefix, we can do it on keg level, like I do that in `Formula#latest_head_prefix`.